dashboard/app/models/user_script.rb

```diff
@@ -36,4 +38,17 @@ def empty?
     # a user script is empty if there is no progress
     started_at.nil? && completed_at.nil? && assigned_at.nil? && last_progress_at.nil?
   end
+
+  def check_plc_task_assignment_updating
+    if check_completed? && self.script.pd
```

Comment:
`UserScript#check_completed?` triggers an extra db query, so triggering this on every user-script update on any script would be a performance/scalability regression we don't want to introduce. `self.script.pd` does not trigger any db query, so if we reverse the conditional order here we should limit this extra query only to PD levels, which would be acceptable